File: php/src/Resource/TaskComments.php
<?php

declare(strict_types=1);

namespace Fivexer\SDK\Resource;

use Fivexer\SDK\Fivexer;
use Fivexer\SDK\Model\AddComment;
use Fivexer\SDK\Model\Comment;
use Fivexer\SDK\Model\CommentPage;

/** Comments on a task, cursor-paginated newest-first. */
final class TaskComments
{
    public function __construct(private readonly Fivexer $client)
    {
    }

    public function add(string $taskId, AddComment $input): Comment
    {
        $data = $this->client->request('POST', '/tasks/' . \rawurlencode($taskId) . '/comments', $input->toArray()) ?? [];
        return Comment::fromArray($data['comment'] ?? []);
    }

    /**
     * @param string|null $cursor Pagination cursor from a previous page's nextCursor
     * @param int<1, max>|null $limit Page size
     */
    public function list(string $taskId, ?string $cursor = null, ?int $limit = null): CommentPage
    {
        $data = $this->client->request('GET', '/tasks/' . \rawurlencode($taskId) . '/comments', null, [
            'cursor' => $cursor,
            'limit' => $limit,
        ]) ?? [];
        return CommentPage::fromArray($data);
    }

    public function remove(string $taskId, string $commentId): void
    {
        $this->client->request(
            'DELETE',
            '/tasks/' . \rawurlencode($taskId) . '/comments/' . \rawurlencode($commentId)
        );
    }
}

File: java/src/main/java/io/fivexer/sdk/TaskComments.java
package io.fivexer.sdk;

import io.fivexer.sdk.internal.Json;
import io.fivexer.sdk.model.AddComment;
import io.fivexer.sdk.model.Comment;
import io.fivexer.sdk.model.CommentEnvelope;
import io.fivexer.sdk.model.CommentPage;

/** Comments on a task, cursor-paginated newest-first. */
public final class TaskComments {

    private final Fivexer client;

    TaskComments(Fivexer client) {
        this.client = client;
    }

    public Comment add(String taskId, AddComment input) {
        CommentEnvelope envelope = client.request("POST", "/tasks/" + Json.enc(taskId) + "/comments",
                input.toJson(), null, CommentEnvelope.class, false);
        return envelope.getComment();
    }

    public CommentPage list(String taskId, String cursor, Integer limit) {
        return client.request("GET", "/tasks/" + Json.enc(taskId) + "/comments", null,
                Json.query("cursor", cursor, "limit", Json.str(limit)), CommentPage.class, false);
    }

    public CommentPage list(String taskId) {
        return list(taskId, null, null);
    }

    public void remove(String taskId, String commentId) {
        client.request("DELETE", "/tasks/" + Json.enc(taskId) + "/comments/" + Json.enc(commentId),
                null, null, Void.class, true);
    }
}
